### PR TITLE
Merge NOAA-GSL gsl/develop into develop, update all GSL regression tests to use new GFS v16 C96L127 ICs, update macOS build config

### LINKED ISSUE #703: Restart reproducibility broken for GSL suite - fix and add test to rt.conf
## Description

Once again, the updates to the authoritative ufs-weather-model repository and its submodules broke the restart reproducibility for the GSL suite (FV3_GSD_v0). This happens every few months. We need to add restart tests (that already exist in `rt_ccpp_dev.conf`) to `rt.conf` to catch these issues at the time the "offending" code gets tested and merged.

We should consider updating this restart test, which currently runs 0-48h for the continuous run, 0-24h for coldstart and 24-48h for restart (called warmstart; name should be changed, too).

### To Reproduce:
On Hera, with Intel or GNU:
```
./rt.sh -l rt_ccpp_dev.conf -c -e 2>&1 | tee rt_ccpp_dev_create.log
./rt.sh -l rt_ccpp_dev.conf -m -e 2>&1 | tee rt_ccpp_dev_verify.log
```

### LINKED ISSUE #952: Update global regression tests for RAP (formerly GSL) physics
## Description

The GSL regression tests still use the old C96L64 input data from GFS v15. The tests must be updated to use the new C96L127 GFS v16 input data. Also, the tests should all be renamed to remove the institution (GSD, GSL) from the name and base it on an application/model (RAP in this case).

This will be done together with an update to several of the RAP physics from the NOAA-GSL gsl/develop forks of the ufs-weather-model, fv3atm, ccpp-physics.

### PR DESCRIPTION
**Pull request by climbfuji**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [x] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Description

This PR is a major overhaul of the regression tests employing the physics developed at NOAA-GSL. The new control for these suites is now `rap_control` to move away from institutional names. Based on this new control, the analog to the operational requirements tests are added for decomposition, threading, restart etc. Feature tests are added that swap out individual physics one at a time. Most tests are created in DEBUG mode and run only for 1h to speed up the testing.

This PR requires new input data and new baselines. The current C96L127 `gfs_data.tile?.nc` files in the input data need to be replaced with files that contain the Thompson aerosol climatology, but are otherwise identical. This has no impact on model runs that do not use Thompson aerosol-aware microphysics. All new tests use C96L127 instead of the old C96L64 input data. However, I think that there are still some tests (HAFS) that use the old input data, therefore it cannot be removed yet.

The submodule pointer updates for fv3atm and ccpp-physics contain several updates and bugfixes for RAP physics and uite definition files, including a sign error for UGWP, bug fixes in Thompson MP when using the inner loop, updates to RUC LSM, GF convection, MYNN PBL/SFC.

New baselines required. The following tests change results due to updates to the RAP physics (Thompson, UGWP, ...):
```
control_debug_p7
control_p7
control_p7_rrtmgp
control_thompson
control_thompson_debug
control_thompson_debug_extdiag
cpld_bmark_p7
cpld_control_c192_p7
cpld_control_c384_p7
cpld_control_c96_p7
cpld_control_p7_rrtmgp
cpld_debug_p7
fv3_regional_RRTMGP
fv3_regional_control
fv3_regional_debug
fv3_regional_hafs
fv3_regional_netcdf_parallel
fv3_regional_noquilt
```
All old GSL tests are removed:
```
fv3_gsd
fv3_gsd_debug
fv3_gsd_diag_debug
fv3_hrrr
fv3_rap
fv3_rrfs_v1alpha
fv3_rrfs_v1alpha_debug
fv3_rrfs_beta
fv3_rrfs_beta_debug
```
New tests are:
```
hrrr_control
rap_*
rrfs_*
```

For follow-up work, see https://github.com/ufs-community/ufs-weather-model/issues/953.

### New input data
New input data set input-data-20211210 is available on the following machines
- [x] hera.{intel,gnu}
- [x] orion.intel
- [x] cheyenne.{intel,gnu}
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [not necessary] CI
- [x] S3

### Issue(s) addressed

Fixes https://github.com/ufs-community/ufs-weather-model/issues/952
Fixes https://github.com/ufs-community/ufs-weather-model/issues/703

## Testing

Regression tests were run regularly on Hera with Intel and GNU to make sure that only the results that were expected to change do change. Full regression tests will be run on all tier-1 platforms at the time of commit.

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [not necessary] opnReqTest for newly added/changed feature (feature tests added to `rt.conf`)
- [x] CI - https://github.com/ufs-community/ufs-weather-model/pull/937/commits/a8b21759bec3643d052515ace9ce07328d2848dd

## Dependencies

- waiting on https://github.com/NCAR/ccpp-physics/pull/791
- waiting on https://github.com/NOAA-EMC/fv3atm/pull/435

### CODE REVIEW

File: tests/tests/rap_restart [hunk 1]
```diff
@@ -52,50 +32,70 @@ export LIST_FILES="sfcf027.tile1.nc \
...
+                   RESTART/sfc_data.tile2.nc \
+                   RESTART/sfc_data.tile3.nc \
+                   RESTART/sfc_data.tile4.nc \
+                   RESTART/sfc_data.tile5.nc \
+                   RESTART/sfc_data.tile6.nc"
 
 export_fv3
+export NPZ=127
+export NPZP=128
+export DT_ATMOS=720
```
Comment (DeniseWorthen) [inline tests/tests/rap_restart]: In this and most tests dt_atmos is set here and then again later as dt_atmos=300.

File: tests/tests/rap_rrtmgp_debug [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+export OUTPUT_FH="0 1"
+#export WRITE_DOPOST=.true.
+
+export FV3_RUN=control_run.IN
+export CCPP_SUITE=FV3_RAP_RRTMGP
+export INPUT_NML=rap.nml.IN
+export FIELD_TABLE=field_table_thompson_aero_tke
+export NEW_DIAGTABLE=diag_table_rap
+
+export DT_ATMOS="150"
```
Comment (DeniseWorthen) [inline tests/tests/rap_rrtmgp_debug]: here dt_atmos is set to 150? (720 at ln 19)

File: tests/tests/hrrr_control [hunk 3]
```diff
@@ -72,29 +54,39 @@ export LIST_FILES="atmos_4xdaily.tile1.nc \
...
-export FHMAX=24
-export OUTPUT_FH="12 -1"
+export FV3_RUN=control_run.IN
+export CCPP_SUITE=FV3_HRRR
+export INPUT_NML=rap.nml.IN
+export FIELD_TABLE=field_table_thompson_aero_tke
+export NEW_DIAGTABLE=diag_table_rap
 
-export DT_ATMOS="600"
+export DT_ATMOS="300"
```
Comment (DeniseWorthen) [inline tests/tests/hrrr_control]: we can remove the quotes around all the 300 values for dt_atmos

File: tests/tests/rap_2threads [hunk 4]
```diff
@@ -72,26 +54,41 @@ export LIST_FILES="atmos_4xdaily.tile1.nc \
...
+export WRTTASK_PER_GROUP=6
 
-export DT_ATMOS="600"
+export FV3_RUN=control_run.IN
+export CCPP_SUITE=FV3_RAP
+export INPUT_NML=rap.nml.IN
+export FIELD_TABLE=field_table_thompson_aero_tke
+export NEW_DIAGTABLE=diag_table_rap
+
+export DT_ATMOS="300"
```
Comment (DusanJovic-NOAA) [inline tests/tests/rap_2threads]: No need for double quotes.

File: tests/tests/rap_rrtmgp_debug (hunk 2 above)
Comment (climbfuji) [reply tests/tests/rap_rrtmgp_debug]: Yes. The 127 vertical levels really pose a problem for the long time steps in FV3 when using RAP physics plus RRTMGP.

File: tests/tests/hrrr_control (hunk 3 above)
Comment (climbfuji) [reply tests/tests/hrrr_control]: Thanks, Denise. I will remove all duplicates of dt_atmos and the double quotes around the remaining ones.

File: tests/tests/control_diag_debug [hunk 5]
```diff
@@ -31,7 +31,6 @@ export FV3_RUN=control_run.IN
 export CCPP_SUITE=FV3_GFS_v16
 export INPUT_NML=control.nml.IN
 
-export FDIAG=0,1,2,3,4,5,6
```
Comment (DeniseWorthen) [inline tests/tests/control_diag_debug:34]: This is just clean up since fhmax is only 1 hour?

Comment (climbfuji) [reply tests/tests/control_diag_debug:34]: `FDIAG` is no longer used, I think. The control_diag_debug test passes without this setting.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: BL
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/792775797/20211211171514/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_277549
Baseline creation and move successful
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/792775797/20211211194306/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_214854
Test rap_2threads 041 failed failed 
Test rap_2threads 041 failed in run_test failed 
Please make changes and add the following label back:
jet-intel-BL

**Review by DeniseWorthen — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by MinsukJi-NOAA — APPROVED** [review]

**Comment by climbfuji** [comment]
Hash for fv3atm is correct (e7687f4), ready to merge

**Pull request merged**